packages/js-go-feature-flag-provider/src/lib/go-feature-flag.ts

```diff
@@ -0,0 +1,121 @@
+import {ContextTransformer, EvaluationContext, Provider, ResolutionDetails,} from '@openfeature/openfeature-js';
+import {FlagNotFoundError, GeneralError, TypeMismatchError} from '@openfeature/extra';
+import axios from 'axios';
+import {
+  GoFeatureFlagProviderOptions,
+  GoFeatureFlagProxyRequest,
+  GoFeatureFlagProxyResponse,
+  GoFeatureFlagUser,
+} from "./model";
+import {ProxyNotReady} from "./proxy-not-ready";
+
+
+const DEFAULT_CONTEXT_TRANSFORMER = (context: EvaluationContext): GoFeatureFlagUser => {
+  // TODO: rework the context transformer
+  const { targetingKey, ...attributes } = context;
+  return {
+    key: targetingKey || (attributes['userId'] as string) || 'anonymous',
+    anonymous: !targetingKey,
+    custom: attributes
+  }
+};
```

Comment:
Any feedback on the context transformer concept? 

Not all of our SDKs have it, it's optional. Some people liked the idea and some found it a bit confusing, so I'm wondering what you think.

Reply:
I like this pattern even if it requires a bit more documentation to understand what it does in the beginning.
But in the end this is super useful to be able to convert from the `EvaluationContext` to whatever format you need.

The only thing that is a bit disturbing for me is that the `targetingKey` is not mandatory this makes my life harder, but if I understand correctly this is not mandatory on all feature flag solutions.

Reply:
> but if I understand correctly this is not mandatory on all feature flag solutions.

Correct. There's other reasons too, but they are tough to explain here.

Reply:
One other feedback but this is not related to the provider but more from the repo, I was not able to make unit tests working.
I am not familiar with `nx` so maybe I missed something but I had weird typescript error with invalid export error.

Reply:
To best honest the test maybe have been broken for some time, since it's not in the CI. I will add an issue for this. 

It's been a low priority since this is just a playground, but we should get them working.